host/installer/installer.nsi

```diff
@@ -0,0 +1,196 @@
+	!define line4 "   (Must be uppercase, the last option given and no quotes)$\r$\n$\r$\n"
+	!define line5 "/startmenu - create start menu shortcut$\r$\n"
+	!define line6 "/desktop - create desktop shortcut$\r$\n"
+    !define line7 "/noservice - do not create a service to auto start and elevate the host"
+	MessageBox MB_OK "${line1}${line2}${line3}${line4}${line5}${line6}${line7}"
+	Abort
+FunctionEnd
+
+Function .onInit
+
+    var /GLOBAL cmdLineParams
+    Push $R0
+    ${GetParameters} $cmdLineParams
+    ClearErrors
+	
+    ${GetOptions} $cmdLineParams '/?' $R0
+    IfErrors +2 0
+    Call ShowHelpMessage
+	
+	${GetOptions} $cmdLineParams '/H' $R0
+    IfErrors +2 0
+    Call ShowHelpMessage
+	
+    Pop $R0
+
+
+    Var /GLOBAL option_startMenu
+    Var /GLOBAL option_desktop
+    Var /GlOBAL option_noservice
+    StrCpy $option_startMenu     0
+	StrCpy $option_desktop       0
+    StrCpy $option_noservice     0
+
+    Push $R0
+		
+	${GetOptions} $cmdLineParams '/startmenu' $R0
+    IfErrors +2 0
+	StrCpy $option_startMenu 1
+
+    ${GetOptions} $cmdLineParams '/desktop' $R0
+    IfErrors +2 0
+    StrCpy $option_desktop 1
+    
+    ${GetOptions} $cmdLineParams '/noservice' $R0
+    IfErrors +2 0
+    StrCpy $option_noservice 1
+    
+    Pop $R0
+
+FunctionEnd
+
+;Install 
+Section "-Install" Section1
+
+  SetOutPath $INSTDIR
+  File ..\build\looking-glass-host.exe
+  File /oname=LICENSE.txt ..\..\LICENSE
+  WriteUninstaller $INSTDIR\uninstaller.exe
+
+  ${GetSize} "$INSTDIR" "/S=0K" $0 $1 $2
+  IntFmt $0 "0x%08X" $0
+  WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"EstimatedSize" "$0"
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"DisplayName" "Looking-Glass"
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"UninstallString" "$\"$INSTDIR\uninstaller.exe$\""
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"QuietUninstallString" "$\"$INSTDIR\uninstaller.exe$\" /S"
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"InstallLocation" "$INSTDIR"
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"Publisher" "Geoffrey McRae"
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"DisplayIcon" "$\"$INSTDIR\looking-glass-host.exe$\""
+  WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"NoRepair" "1"
+  WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+	"NoModify" "1"
+  WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Looking-Glass" \
+    "DisplayVersion" ${VERSION}
+
+SectionEnd
+
+Section "Looking-Glass Service" Section2
+
+  ${If} $option_noservice == 0
+    nsExec::Exec '"$INSTDIR\looking-glass-host.exe" UninstallService'
```

Comment:
This fails if the service is still running, you must stop it first. See `https://nsis-dev.github.io/NSIS-Forums/html/t-293237.html` for an example.

Reply:
Scratch this, I am adding this to the host application itself.